### ISSUE TITLE
Incorrect commit_date storing in SQL DB

### ISSUE THREAD

**Issue by floresek** [javers/javers#325]
I noticed that 31-12-2015 (13f476be06c9d50227073cb18c54c2560d6eaec3) you changed storing of jv_commit.commit_date from local timestamp to ?

I guess that your intention was to store commit time in UTC time zone but:
``` java
// class org.javers.repository.sql.repositories.CommitMetadataRepository

    private Timestamp toTimestamp(LocalDateTime commitMetadata) {
        return new Timestamp(commitMetadata.toDate(TimeZone.getTimeZone("UTC")));
    }
```
gives the time which would be in defult time zone (local) if passed (commitMetadata) time will be set in UTC time zone.
F.e. in Europe/Belgrade local time 18:00 is converted to 'UTC' 19:00 but should be to: 17:00

Proper solution should be to create adequate DB colum type (f.e. TIMESTAMP WITH TIME ZONE in Oracle)
and set column value in different way (not resulting to JDBC setTimestamp(idx, ts) as currently is).

JDBC specification suggests using  setTimestamp(idx, ts, calendar) - (ts local; calendar in UTC time zone)
but in Oracle it causes saving default time zone instead of UTC (with offsetting the time valid for UTC).

Correct result can be achieved by passing string - date converted to UTC f.e.:

``` groovy
def c = Calendar.getInstance(TimeZone.getTimeZone('UTC'))
def sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss z")
sdf.setTimeZone(TimeZone.getTimeZone('UTC'))
def stmt = conc.prepareStatement('INSERT INTO TEST (ts_with_tz) VALUES(?)')
def ts = new Timestamp(new LocalDateTime().toDate().getTime())
stmt.setString(1, sdf.format(ts))
```

If you want to to store UTC offsetted time in TIMESTAMP column you can use f.e.:
``` java
    private Timestamp toTimestamp(LocalDateTime commitMetadata) {
        return new Timestamp(DateTimeZone.getDefault().convertLocalToUTC(commitMetadata.toDate().getTime(), true))
    }
```

By the way - doing such a change you should warn in release notes that conversion of existing jv_commit.commit_date data is required!
(and provide update scripts)

Cheers

**Comment by bartoszwalacik** [comment]
ouch, you are right. Think we dont want to deep dive into Oracle calendars ...
We will change SqlRepository to store commitDate as is (so as local datetime)

**Comment by bartoszwalacik** [comment]
take a look at my commit

**Comment by floresek** [comment]
ok - but I think that storing commit date in UTC time zone should be applied in javers - especially that you recently added .from(date) .to(date) filters to repository (which will not work property for daylight saving change periods).

**Comment by bartoszwalacik** [comment]
Well, I dont think that storing commitDate in UTC is a good idea. We should strive to use local time.
It's much easier to understand and manipulate your data using raw SQL when time is not shifted to UTC.

For now I dont know what to do with daylight saving cornercases, we should investigate it more...

**Comment by floresek** [comment]
ok - could you release new version with UTC bug fix, please?

**Comment by bartoszwalacik** [comment]
fix released in JaVers 1.4.10
